### ISSUE TITLE
VC++ 2010 and earlier don't define snprintf

### ISSUE THREAD

**Issue by utelle** [jmcnamara/libxlsxwriter#58]
Unfortunately MSVC++ 2010 and earlier don't define the function `snprintf` ... and what's worse, the function `_snprintf` provided by MSVC++ 2010 can't be used as a replacement, since it behaves differently than `snprintf`. One way to solve the problem is to define a function with the proper behaviour in the header file `common.h` of `libxlsxwriter`. After line 235 add the following code:
```
/* VC++ 2010 doesn't define snprintf */
#if defined(_MSC_VER) && _MSC_VER < 1900

#define snprintf c99_snprintf
#define vsnprintf c99_vsnprintf

#include <stdarg.h>

__inline int c99_vsnprintf(char *outBuf, size_t size, const char *format, va_list ap)
{
    int count = -1;

    if (size != 0)
        count = _vsnprintf_s(outBuf, size, _TRUNCATE, format, ap);
    if (count == -1)
        count = _vscprintf(format, ap);

    return count;
}

__inline int c99_snprintf(char *outBuf, size_t size, const char *format, ...)
{
    int count;
    va_list ap;

    va_start(ap, format);
    count = c99_vsnprintf(outBuf, size, format, ap);
    va_end(ap);

    return count;
}

#endif 
```
This makes `libxlsxwriter` operational with MSVC++ 2010 and earlier.

**Comment by jmcnamara** [comment]
Added to master.

**Comment by jmcnamara** [comment]
Fixed in release 0.4.0. Thanks for the report.
